### PR TITLE
Introduce libkubevirt and reduce cyclic dependency

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
- Introduce a new `libkubevirt` pkg
- Reduce cyclic dependency removing testsuite dependency from clientcmd

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
/cc @EdDev 
Your feedback is appreciated, thanks

File: tests/template_test.go [hunk 1]
```diff
@@ -154,12 +152,12 @@ var _ = Describe("[Serial][sig-compute]Templates", Serial, decorators.SigCompute
 				for param, value := range templateParams {
 					ocProcessCommand = append(ocProcessCommand, "-p", fmt.Sprintf("%s=%s", param, value))
```
Comment (0xFelix) [inline tests/template_test.go:153]: Are these still running? IIRC we wanted to drop `oc` specific tests?

File: tests/virtctl/credentials.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ import (
 	"os"
 	"path/filepath"
 
+	"kubevirt.io/kubevirt/tests/testsuite"
```
Comment (0xFelix) [inline tests/virtctl/credentials.go:8]: Import order?

File: tests/virtctl/scp.go [hunk 3]
```diff
@@ -6,6 +6,8 @@ import (
 	"os"
 	"path/filepath"
 
+	"kubevirt.io/kubevirt/tests/testsuite"
```
Comment (0xFelix) [inline tests/virtctl/scp.go:9]: Import order?

File: tests/virtctl/ssh.go [hunk 4]
```diff
@@ -6,6 +6,8 @@ import (
 	"os"
 	"path/filepath"
 
+	"kubevirt.io/kubevirt/tests/testsuite"
```
Comment (0xFelix) [inline tests/virtctl/ssh.go:9]: Import order?

File: tests/vmi_cloudinit_hook_sidecar_test.go [hunk 5]
```diff
@@ -25,6 +25,8 @@ import (
 	"path/filepath"
 	"time"
 
+	"kubevirt.io/kubevirt/tests/testsuite"
```
Comment (0xFelix) [inline tests/vmi_cloudinit_hook_sidecar_test.go:28]: Import order?

File: tests/vmi_hostdev_test.go [hunk 6]
```diff
@@ -5,14 +5,14 @@ import (
 	"fmt"
 	"strings"
 
+	"kubevirt.io/kubevirt/tests/testsuite"
```
Comment (0xFelix) [inline tests/vmi_hostdev_test.go:8]: Import order?

File: tests/vmi_sound_test.go [hunk 7]
```diff
@@ -24,6 +24,8 @@ import (
 	"encoding/xml"
 	"fmt"
 
+	"kubevirt.io/kubevirt/tests/testsuite"
```
Comment (0xFelix) [inline tests/vmi_sound_test.go:27]: Import order?

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter_test.go [hunk 8]
```diff
@@ -24,6 +24,8 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"kubevirt.io/kubevirt/tests/testsuite"
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter_test.go:27]: Nice catch. It was already there but it is good to drop the dependency here

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @fossedihelm.
Could you please give a few words about how this new package differs from `tests/libkvconfig`?
Does it make sense to merge them?

File: tests/template_test.go (hunk 1 above)
Comment (fossedihelm) [reply tests/template_test.go:153]: Actually, IDK about this but if you agree I would leave this for a follow-up PR.
This PR already has a huge context and I prefer not to increase it more. WDYT?
@xpivarc Are you aware of the removal of `oc` stuff?

**Comment by fossedihelm** [comment]
@0xFelix I addressed the comments (except the `oc` ones). PTAL

**Comment by fossedihelm** [comment]
> Thank you for the PR @fossedihelm. Could you please give a few words about how this new package differs from `tests/libkvconfig`? Does it make sense to merge them?

@orelmisan I would prefer to leave them separated for the moment, since I have the impression that there could be several thing that can be added to `config` or `libkubevirt`. (i.e. `UpdateKubeVirtConfigValueAndWait`, `DisableFeatureGate`, `EnableFeatureGate`, etc...) If this will not be true we can think about merging them later. WDYT?

**Comment by fossedihelm** [comment]
@vasiliy-ul FYI

**Comment by orelmisan** [comment]
> > Thank you for the PR @fossedihelm. Could you please give a few words about how this new package differs from `tests/libkvconfig`? Does it make sense to merge them?
> 
> @orelmisan I would prefer to leave them separated for the moment, since I have the impression that there could be several thing that can be added to `config` or `libkubevirt`. (i.e. `UpdateKubeVirtConfigValueAndWait`, `DisableFeatureGate`, `EnableFeatureGate`, etc...) If this will not be true we can think about merging them later. WDYT?

My feedback was mostly around providing reasoning for why things were done.
Since there is an existing package with a similar name, I find it helpful to explain why is there a need to add another package, what would go in it etc.

File: tests/template_test.go (hunk 1 above)
Comment (xpivarc) [reply tests/template_test.go:153]: We don't have a lane for it, so yes it would be reasonable to remove.

Comment (fossedihelm) [reply tests/template_test.go:153]: :+1: Will check in a follow up PR. Thank you

**Comment by fossedihelm** [comment]
@orelmisan Added a description in the commits, was that what you mean? Thank you! :)

**Comment by fossedihelm** [comment]
Thank you @orelmisan! I would try to give it another chance. In case of another rebase I will split it into separate PRs! :)

File: tests/performance/density.go [hunk 9]
```diff
@@ -26,27 +26,24 @@ import (
...
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	audit_api "kubevirt.io/kubevirt/tools/perfscale-audit/api"
-	metric_client "kubevirt.io/kubevirt/tools/perfscale-audit/metric-client"
-
 	kvv1 "kubevirt.io/api/core/v1"
 	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+	"kubevirt.io/client-go/kubecli"
 
 	instancetypeBuilder "kubevirt.io/kubevirt/tests/libinstancetype/builder"
```
Comment (0xFelix) [inline tests/performance/density.go:38]: nit: Imports here are still a bit off

**Review by 0xFelix — COMMENTED** [review]
Looks good to me, just a single nit.

File: tests/performance/density.go (hunk 9 above)
Comment (fossedihelm) [reply tests/performance/density.go:38]: image

**Comment by fossedihelm** [comment]
@0xFelix PTAL

**Review by 0xFelix — APPROVED** [review]
Thanks!

/lgtm

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Comment by Barakmor1** [comment]
I agree with @orelmisan . This PR does too many things and should be split into smaller PRs. This will make it easier to review and reduce the likelihood of needing rebases.

/sig code-quality

**Comment by fossedihelm** [comment]
@orelmisan @Barakmor1 It is not so easy to split this. Consider that the cyclic dependency is resolved only with the last commits and the first commits does not compile without the others.

**Review by 0xFelix — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**